.github/workflows/major-version-approval.yaml

```
@@ -0,0 +1,53 @@
+name: Major Version Approval Check
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, edited]
+    branches: ['main']
+  pull_request_review:
+    types: [submitted]
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  major-version-approval-check:
+    name: Major Version Approval Check
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.user.login == 'release-please[bot]' && contains(github.event.pull_request.body, 'MAJOR_VERSION_ALLOWED=')
+    steps:
```

we need to add the checkout step here so that the git repo is available


```suggestion
    steps:
      - uses: actions/checkout@v5
```